server/liveblog/client_modules/client_modules.py

```diff
@@ -557,6 +510,59 @@ def get_blog_posts(blog_id):
     return api_response(result_data, 200)
 
 
+@voting_blueprint.route("/api/client_poll_vote/<poll_id>", methods=["POST"])
+def client_poll_vote(poll_id):
+    """
+    Handles the voting action for a poll.
+
+    This function allows users to vote for a specific option in a poll. It performs the following steps:
+    1. Validate the request data
+    2. Retrieve the poll and validate selected option
+    3. Increment the vote count
+    4. Update associated blog posts to reflect change on embed
+
+    Returns:
+        JSON response with code:
+            - 201: If the vote is successfully recorded.
+            - 400: If the request is invalid or any check fails.
+            - 500: If an error occurs during vote update.
+
+    **TODO:**
+    * Add rate limiter decorator on endpoint
+    """
+    data = request.json
+    option_selected = data.get("option_selected", None)
+    if option_selected is None:
+        return api_error("Please select a voting option.", 400)
+
+    polls = get_resource_service("polls")
+    poll = polls.find_one(req=None, _id=poll_id)
+    if poll is None:
+        return api_error("Error: Poll not found", 404)
+
+    poll_body = poll.get("poll_body", {})
+    answers = poll_body.get("answers", [])
+    option_exists = any(answer.get("option") == option_selected for answer in answers)
```

Comment:
this piece got into my mind. Depending on the number of the answers coming from the frontend, this loop could be kinda tricky but we'll leave it like this for now. Once we have the rate limiter it should not be a problem anyways.